src/components/Editor/Footer.js

```diff
@@ -248,12 +248,18 @@ class SourceFooter extends PureComponent<Props, State> {
 
 const mapStateToProps = state => {
   const selectedSource = getSelectedSource(state);
-  const selectedId = selectedSource.id;
+  let prettySource = null;
+  let mappedSource = null;
+
+  if (selectedSource) {
+    prettySource = getPrettySource(state, selectedSource.id);
+    mappedSource = getGeneratedSource(state, selectedSource);
```

Comment:
i wonder if we can change these functions to just return null if there is no source

Reply:
I think the downside of that is that it means the return value has to be nullable, so even if we know the input exists, Flow will end up requiring us to assert that the return value is also not nullable.